### ISSUE TITLE
Compile fails with threaded perls

### ISSUE THREAD

**Issue by eserte** [perlpunk/YAML-LibYAML-API-p5#1]
Compilation fails with threaded perls:
```
cc -c  -I.  -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_FORTIFY_SOURCE=2 -O2 -pipe    -DVERSION=\"0.000\"  -DXS_VERSION=\"0.000\" -DPIC -fPIC "-I/usr/perl5.22.2t/lib/5.22.2/amd64-freebsd-thread-multi/CORE"   XS.c
In file included from XS.xs:11:
./etc/perl_libyaml.c:55:18: error: use of undeclared identifier 'my_perl'
    perl_event = newHV();
                 ^
/usr/perl5.22.2t/lib/5.22.2/amd64-freebsd-thread-multi/CORE/hv.h:627:28: note: expanded from macro 'newHV'
#define newHV() MUTABLE_HV(newSV_type(SVt_PVHV))
                           ^
(... etc ...)
```

**Comment by perlpunk** [comment]
Thanks, you're fast ;-)
any hints appreciated on how to solve this...

**Comment by eserte** [comment]
It's this aTHX/dTHX/... magic what's missing here... maybe something like this will solve it:
```
diff --git a/LibYAML/etc/perl_libyaml.c b/LibYAML/etc/perl_libyaml.c
index a70d45b..c60705f 100644
--- a/LibYAML/etc/perl_libyaml.c
+++ b/LibYAML/etc/perl_libyaml.c
@@ -39,6 +39,7 @@ parser_error_msg(yaml_parser_t *parser, char *problem)
 HV *
 libyaml_to_perl_event(yaml_event_t *event)
 {
+    dTHX;
     HV *perl_event;
     HV *perl_start_mark;
     HV *perl_end_mark;
```

**Comment by perlpunk** [comment]
Indeed, that solves it! Just in time, my installation of a threaded perl just finished ;-)

**Comment by perlpunk** [comment]
Fixed and uploaded 0.001_001
